### PR TITLE
Implement bottleneck detection for workflow monitoring

### LINKED ISSUE #101: [P1] Implement LeadAgent.detect_bottlenecks() logic
## Overview
Bottleneck detection is critical for optimizing multi-agent workflows, but the method is currently stubbed.

## Current State
**File**: `codeframe/agents/lead_agent.py` lines 594-596

```python
def detect_bottlenecks(self) -> list:
    """Detect workflow bottlenecks."""
    # TODO: Implement bottleneck detection
    return []
```

## Expected Behavior
Detect bottlenecks when:
- Tasks are waiting on dependencies longer than threshold
- Agents are idle while tasks are blocked
- Specific agents are overloaded with too many assignments
- Critical path tasks are delayed

## Implementation Requirements
1. Query database for tasks with status='blocked' or 'waiting'
2. Calculate wait times for each blocked task
3. Identify agents with excessive workload (>5 tasks)
4. Identify idle agents (0 tasks assigned)
5. Detect dependency chain delays
6. Return list of bottleneck descriptors with severity

## Return Format
```python
[
    {
        "type": "dependency_wait",
        "task_id": 27,
        "wait_time_minutes": 120,
        "blocking_task_id": 25,
        "severity": "high"
    },
    {
        "type": "agent_overload",
        "agent_id": "backend-001",
        "assigned_tasks": 8,
        "severity": "medium"
    }
]
```

## Acceptance Criteria
- [ ] Bottlenecks detected for dependency waits
- [ ] Bottlenecks detected for agent overload/underload
- [ ] Severity levels assigned correctly
- [ ] Recommendations provided for resolution
- [ ] Unit tests verify detection logic
- [ ] Integration tests verify bottleneck reporting

## Priority
**P1 - Critical**: Required for efficient multi-agent coordination and preventing workflow stalls.

## References
- User Documentation: CODEFRAME_USER_DOCUMENTATION.md lines 277-278
- Issues Analysis: CODEFRAME_ISSUES_ANALYSIS.md lines 31-38

### PR DESCRIPTION
**Pull request by frankbria**
## Summary
Implements `detect_bottlenecks()` method to identify workflow performance issues in multi-agent coordination. Provides real-time detection of 4 bottleneck types with severity classification and actionable recommendations.

## Closes
Closes #101

### Core Features
- **4 Bottleneck Types**:
  1. **Dependency Wait**: Tasks blocked >60min waiting on dependencies
  2. **Agent Overload**: Agents with excessive task assignments (future-ready for queue-based architecture)
  3. **Agent Idle**: Idle agents while pending tasks exist (resource underutilization)
  4. **Critical Path**: Tasks blocking ≥3 dependent tasks

- **Severity Classification**: Critical, High, Medium, Low based on configurable thresholds
- **Actionable Recommendations**: Human-readable suggestions for each bottleneck
- **Comprehensive Error Handling**: Graceful degradation, won't crash coordination loop

### Configuration Constants
```python
DEPENDENCY_WAIT_THRESHOLD_MINUTES = 60  # Flag tasks blocked >60min
AGENT_OVERLOAD_THRESHOLD = 5            # Max tasks per agent
CRITICAL_PATH_THRESHOLD = 3             # Min dependents to flag
CRITICAL_SEVERITY_WAIT_MINUTES = 120    # Critical severity threshold
HIGH_SEVERITY_WAIT_MINUTES = 60         # High severity threshold
```

### Test Coverage
- ✅ **38 unit tests** (`tests/agents/test_bottleneck_detection.py`)
  - Helper method tests (calculate wait time, workload, severity, recommendations)
  - Full workflow tests (all 4 bottleneck types)
  - Edge cases (empty state, exceptions, boundary conditions)

- ✅ **3 integration tests** (`tests/agents/test_multi_agent_integration.py`)
  - End-to-end detection with real components
  - Detection during active coordination loop
  - Performance testing (100 tasks, 10 agents, <1s)

- ✅ **All 41 tests passing (100%)**

### Performance Validation
- Detection time: <1 second for 100+ tasks
- Linear complexity: O(n) for tasks, O(m) for agents
- No performance degradation in coordination loop

### Code Review Completed
- ✅ Comprehensive code review (`docs/code-review/2025-12-17-bottleneck-detection-review.md`)
- ✅ Fixed critical workload calculation bug
- ✅ Fixed KeyError risk in agent idle detection
- ✅ All issues resolved, production-ready

### Strengths
- Excellent error handling (all methods wrapped in try-except)
- Clean code structure (single responsibility, clear naming)
- Comprehensive docstrings
- Performance optimized (early exits, no nested loops)

### User Guide
- **Location**: `docs/BOTTLENECK_DETECTION.md`
- **Contents**: 
  - Overview and configuration
  - 4 bottleneck type descriptions with examples
  - API usage patterns
  - Integration patterns for monitoring
  - Troubleshooting guide

### Code Review Report
- **Location**: `docs/code-review/2025-12-17-bottleneck-detection-review.md`
- **Contents**:
  - Security assessment (low risk - internal monitoring)
  - Reliability analysis (excellent error handling)
  - Performance analysis (optimized for 100+ tasks)
  - Issues found and resolved

## Files Changed
```
codeframe/agents/lead_agent.py                          | +267 lines
tests/agents/test_bottleneck_detection.py               | +500 lines (new)
tests/agents/test_multi_agent_integration.py            | +240 lines
docs/BOTTLENECK_DETECTION.md                            | +300 lines (new)
docs/code-review/2025-12-17-bottleneck-detection-review.md | +308 lines (new)
```

## Example Usage

```python
# In LeadAgent coordination loop
bottlenecks = self.detect_bottlenecks()

for bn in bottlenecks:
    if bn["severity"] in ["critical", "high"]:
        logger.warning(f"{bn['severity'].upper()}: {bn['type']} - {bn['recommendation']}")
```

**Sample Output**:
```
WARNING: HIGH: dependency_wait - Task 42 has been waiting 90min on task 12. Investigate or manually unblock dependency.
WARNING: CRITICAL: critical_path - Task 5 blocks 7 dependent tasks. Prioritize this task or parallelize blocking dependencies.
```

## Checklist
- [x] Implementation complete with all helper methods
- [x] 41 tests passing (38 unit + 3 integration)
- [x] Code review completed and issues resolved
- [x] Documentation complete (user guide + review report)
- [x] Performance validated (<1s for 100+ tasks)
- [x] No breaking changes
- [x] Ready for merge

## Risk Assessment
**Risk Level**: LOW
- Internal monitoring function (no external input)
- Comprehensive error handling (graceful degradation)
- 100% test coverage with integration tests
- Performance validated at scale

---

**Reviewer Notes**: This implementation provides real-time workflow visibility for the multi-agent coordination system. The agent overload detection is future-ready for when agents support task queues (currently agents process one task at a time).

## Summary by CodeRabbit

* **New Features**
  - Added workflow bottleneck detection to identify dependency delays, agent overload, idle agents, and critical-path blockages; computes severity and generates actionable recommendations.

* **Documentation**
  - Added comprehensive guide with usage examples, configuration, monitoring, and troubleshooting.

* **Bug Fixes**
  - Fixed agent workload calculation to avoid inflated overload detections and made idle detection more robust.

* **Tests**
  - Added extensive unit and integration tests covering detection logic, severities, recommendations, and performance.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: tests/agents/test_bottleneck_detection.py [hunk 1]
```diff
@@ -0,0 +1,500 @@
...
+        lead_agent.agent_pool_manager.get_agent_status.return_value = {
+            "agent-1": {"status": "busy", "tasks_completed": 3}  # 4 total (below threshold of 5)
+        }
+        lead_agent.dependency_resolver.get_blocked_tasks.return_value = {}
+        lead_agent.dependency_resolver.dependents = {}
+
+        bottlenecks = lead_agent.detect_bottlenecks()
+
+        # Should not have any agent_overload bottlenecks
+        assert not any(bn["type"] == "agent_overload" for bn in bottlenecks)
```
[the remark this answers is not in the document]
Comment (frankbria) [reply tests/agents/test_bottleneck_detection.py]: prompted

File: codeframe/agents/lead_agent.py [hunk 2]
```diff
@@ -673,10 +681,266 @@ def assign_task(self, task_id: int, agent_id: str) -> None:
...
+            if not created_at_str:
+                return 0
+
+            created_at = datetime.fromisoformat(created_at_str)
+            now = datetime.now()
+            wait_minutes = int((now - created_at).total_seconds() / 60)
+            return max(0, wait_minutes)
+        except (ValueError, TypeError) as e:
+            logger.warning(f"Failed to calculate wait time for task {task.get('id')}: {e}")
+            return 0
```
[the remark this answers is not in the document]
Comment (frankbria) [reply codeframe/agents/lead_agent.py]: prompted
